src/udiskslinuxdriveata.c

```diff
@@ -434,7 +434,26 @@ selftest_status_to_string (SkSmartSelfTestExecutionStatus status)
   return ret;
 }
 
-static gboolean get_pm_state (UDisksLinuxDevice *device, GError **error, guchar *count)
+/**
+ * udisks_linux_drive_ata_is_active:
+ * @device: The ata device to get pm state from
+ * @error: Return location for error or %NULL
+ * @state: hex value of pm state
+ *
+ * Get the current power mode status from a drive.
+ *
+ * The format of @state is the result obtained from sending the
+ * ATA command CHECK POWER MODE to the drive.
+ * Known values include:
+ *     0x00 Standby
+ *     0x80 Idle
+ *     0xff Active/Idle
+ * Typically user interfaces will report "Drive is spun down" if @state is
+ * 0x00 and "Drive is spun up" otherwise.
+ *
+ * Returns: %TRUE if the operation succeeded, %FALSE otherwise.
+ */
+gboolean udisks_linux_drive_ata_is_active (UDisksLinuxDevice *device, GError **error, guchar *state)
```

Comment:
I may have explained myself badly, sorry, but I meant `get_pm_state` would be one function and `is_active` would be another. I quite like the combined version you have here too, but then it should return whether the device is active or not with `state` containing the detailed info (state). There would still be `error` to check for errors.

Reply:
See my other concern about combining result values. It's better to have second function (or macro even) to decode active state from the numerical value.